src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/WebWorker.cs

```diff
@@ -0,0 +1,123 @@
+                    {
+                        parentContext.Post((_) => {
+                            if (res.IsFaulted)
+                                tcs.SetException(res.Exception);
+                            else if (t.IsCanceled)
+                                tcs.SetCanceled();
+                            else
+                                tcs.SetResult(res.Result);
+                        }, null);
+                        JSHostImplementation.UninstallWebWorkerInterop();
+                    }, childScheduler);
+                }
+                catch (Exception e)
+                {
+                    Environment.FailFast("WebWorker.RunAsync failed", e);
+                }
+
+            });
+            JSHostImplementation.SetHasExternalEventLoop(t);
+            t.Start();
+            return tcs.Task;
+        }
+
+        public static Task RunAsyncVoid(Func<Task> body)
+        {
+            var parentContext = SynchronizationContext.Current ?? new SynchronizationContext();
+            var tcs = new TaskCompletionSource();
+            var capturedContext = SynchronizationContext.Current;
+            var t = new Thread(() =>
+            {
+                try
+                {
+                    JSHostImplementation.InstallWebWorkerInterop(true);
+                    var childScheduler = TaskScheduler.FromCurrentSynchronizationContext();
+                    Task res = body();
+                    // This code is exiting thread main() before all promises are resolved.
+                    // the continuation is executed by setTimeout() callback of the thread.
+                    res.ContinueWith(t =>
+                    {
+                        parentContext.Post((_) => {
+                            if (res.IsFaulted)
+                                tcs.SetException(res.Exception);
+                            else if (t.IsCanceled)
+                                tcs.SetCanceled();
+                            else
+                                tcs.SetResult();
+                        }, null);
+                        JSHostImplementation.UninstallWebWorkerInterop();
+                    }, childScheduler);
+                }
+                catch (Exception e)
+                {
+                    Environment.FailFast("WebWorker.RunAsync failed", e);
+                }
+
+            });
+            JSHostImplementation.SetHasExternalEventLoop(t);
+            t.Start();
+            return tcs.Task;
+        }
+
+        public static Task Run(Action body)
+        {
+            var parentContext = SynchronizationContext.Current ?? new SynchronizationContext();
+            var tcs = new TaskCompletionSource();
+            var capturedContext = SynchronizationContext.Current;
+            var t = new Thread(() =>
+            {
+                try
+                {
+                    JSHostImplementation.InstallWebWorkerInterop(false);
+                    body();
```

Comment:
In this one, you definitely should uninstall from a `finally` in case `body` throws`